Vostok.Metrics/Primitives/Timer/ITimerExtensions_Measurement.cs

```diff
@@ -14,6 +15,58 @@ public static void Report([NotNull] this ITimer timer, TimeSpan value)
         public static IDisposable Measure([NotNull] this ITimer timer)
             => new Measurement(timer);
 
+        public static T Measure<T>([NotNull] this ITimer timer, Func<T> action)
+        {
+            var stopwatch = Stopwatch.StartNew();
+            try
+            {
+                return action();
+            }
+            finally
+            {
+                timer.Report(stopwatch.Elapsed);
+            }
+        }
+
+        public static void Measure([NotNull] this ITimer timer, Action action)
+        {
+            var stopwatch = Stopwatch.StartNew();
+            try
+            {
+                action();
+            }
+            finally
+            {
+                timer.Report(stopwatch.Elapsed);
+            }
+        }
+
+        public static async Task<T> Measure<T>([NotNull] this ITimer timer, Func<Task<T>> action)
```

Comment:
> Async overloads don't have ...Async suffix in the function name because in this case the user of the library can misspell function call and get wrong measurements (in the case when sync version of the method was called).

я бы наверно всё же добавил суффикс, ведь точно так же ты можешь забыть написать 

**await** Measure(...)

Reply:
Меня пугает, что ты можешь написать `var result = await timer.Measure(async () => await Do())` и код будет работать корректно, но метрика при этом измеряться не будет. Пока что мне кажется, что следование конвенции только создает лишние проблемы и ничем не помогает.

Reply:
Ага, видимо проблема с общими перегрузками в том, что без явной типизации у лямбды () => throw new Exception() выводится тип `Func<Task>` как более специфичный и вызывается не так перегрузка...((

Reply:
Разделил перегрузки, но у синхронной добавил явный суффикс `MeasureSync`.